include/Reader/reader.h

```diff
@@ -3279,6 +3295,105 @@ class ReaderBase {
+  virtual IRNode *vectorMul(IRNode *Vector1, IRNode *Vector2) = 0;
+  virtual IRNode *vectorDiv(IRNode *Vector1, IRNode *Vector2) = 0;
+
+  virtual IRNode *vectorEqual(IRNode *Vector1, IRNode *Vector2) = 0;
+  virtual IRNode *vectorNotEqual(IRNode *Vector1, IRNode *Vector2) = 0;
+
+  /// \brief Return IRNode* Result of UnOp.
+  ///
+  /// \param Vector  argument for UnOp.
+  /// \returns an IRNode representing the result of UnOp 
+  /// or nullptr if UnOp is not supported.
+  virtual IRNode *vectorAbs(IRNode *Vector) = 0;
+  virtual IRNode *vectorSqrt(IRNode *Vector) = 0;
+
+  /// \brief Return result of ctor operation on SIMD Vector Types.
+  ///
+  /// \param ArgsCount Number of arguments on stack for call.
+  /// \param Class The class handle for the call target method's class.
+  /// \param Opcode Operation Opcode to distinguish newobj from ctor.
+  /// \returns an IRNode representing the result of ctor 
+  /// or nullptr if ctor is not supported.
+  IRNode *generateSMIDCtor(CORINFO_CLASS_HANDLE Class, int ArgsCount, ReaderBaseNS::CallOpcode Opcode);
+
+  /// \brief Return IRNode* Result of ctor.
+  ///
+  /// \param Class The class handle for the call target method's class.
+  /// \param This in not null for ctor, null for newobj.
+  /// \param Args, args for ctor.
+  /// \returns an IRNode representing the result of ctor 
+  /// or nullptr if ctor is not supported.
+  virtual IRNode *vectorCtor(CORINFO_CLASS_HANDLE Class, IRNode *This, std::vector<IRNode *> Args) = 0;
+  
+  /// \brief Return length of Generic Vector.
+  ///
+  /// \param Class The class handle for the call target method's class.
+  /// \returns an IRNode representing the length of Class
+  /// or nullptr if getCount or Class is not supported.
+  virtual IRNode *vectorGetCount(CORINFO_CLASS_HANDLE Class) = 0;
+  
+  /// \brief Return IRNode* The result of the intrinsic or nullptr, if it is unnsupported.
+  ///
+  /// \param Class The class handle for the call target method's class.
+  /// \param  Method handle for the target Method.
+  /// \param  SigInfo info for the target Method.
+  /// \returns an IRNode representing the result of the intrinsic 
+  /// \or nullptr if the intrinsic is not supported.
+  IRNode *generateSIMDIntrinsicCall(CORINFO_CLASS_HANDLE Class,
+                                    CORINFO_METHOD_HANDLE Method,
+                                    CORINFO_SIG_INFO *SigInfo,
+                                    ReaderBaseNS::CallOpcode Opcode);
+
+  /// \brief Check LLVM::VectorType.
+  ///
+  /// \param Arg The target for checking.
+  /// \returns true if Arg is supported vector type.
+  virtual bool checkVectorType(IRNode *Arg) = 0;
+
+  /// \brief Return length of Vector or 0 if it is not Vector.
+  ///
+  /// \param Class The class handle for the call target method's class.
+  /// \returns number of elements in vector.
+  virtual int getSizeOfSIMDType(CORINFO_CLASS_HANDLE Class) = 0;
```

Comment:
Nit: it might be better to rename this to `getElementCountOfSIMDType` or similar. I would expect `getSizeOfSIMDType` to return the size in bytes, given the name.